lib/utils.js

```diff
@@ -475,25 +476,17 @@ function applyWriteConcern(target, sources, options) {
   }
 
   if (options.w != null || options.j != null || options.fsync != null) {
-    const writeConcern = {};
-    if (options.w != null) writeConcern.w = options.w;
-    if (options.wtimeout != null) writeConcern.wtimeout = options.wtimeout;
-    if (options.j != null) writeConcern.j = options.j;
-    if (options.fsync != null) writeConcern.fsync = options.fsync;
-    return Object.assign(target, { writeConcern });
+    const writeConcern = WriteConcern.fromOptions(options);
```

Comment:
you no longer need the check on the line before, `WriteConcern.fromOptions` is already doing that and returning `null` if they aren't set